ray-on-gke/tpu/kuberay-tpu-webhook/main.go

```diff
@@ -315,32 +381,26 @@ func getEnvironmentVariable(varName string, container corev1.Container) string {
 
 // gets the  next lowest-index pod slice (worker group replica) to assign a pod to in the RayCluster
 // there are three possible cases here:
-//  1. sliceToWorkers is empty, this is the first pod the webhook intercepts
+//  1. sliceToWorkerIDs is empty, this is the first pod the webhook intercepts
 //     - assign this pod to replica 0
-//  2. The pod slice exists in sliceToWorkers, but has # created workers < NumOfHosts
+//  2. The pod slice exists in sliceToWorkerIDs, but has # created workers < NumOfHosts
 //     - assign this pod to the lowest index replica with # created workers < NumOfHosts
-//     - since we update isCreated when a worker is deleted, this allows us to assign re-created
 //     pods to the same replica
-//  3. sliceToWorkers isn't empty, but all slices have # workers == NumOfHosts
+//  3. sliceToWorkerIDs isn't empty, but all slices have # workers == NumOfHosts
 //     - this occurs when the pod we intercept is the first pod of a different slice in the cluster
-//     - we keep track of how many replicas of the same worker group have been added to sliceToWorkers
+//     - we keep track of how many replicas of the same worker group have been added to sliceToWorkerIDs
 //     so far, and assign this pod to the next integer replicaIndex
-func getReplicaIndex(clusterName string, groupName string, namespace string) int {
+func getReplicaIndex(sliceToWorkerIDs map[slice][]int, clusterName string, groupName string, namespace string) int {
 	// first pod created in cluster
-	if sliceToWorkers == nil {
+	if sliceToWorkerIDs == nil {
```

Comment:
if len(sliceToWorkerIDs) == 0? 

Unless an initliazed, but empty sliceToWorkerIDs should not return 0 here

Reply:
Both those cases would return 0, I'll add this change in.

Reply:
[8106aad](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/723/commits/8106aad2b719212bddcb76e3a5db36b363ed5122)